pkg/apis/apiserver/organization_REST.go

```diff
@@ -336,8 +340,42 @@ func internalErrorWatchEvent(err error) watch.Event {
 	}
 }
 
-func (o *OrganizationREST) isVisible(ctx context.Context, organizationName string, verb string) (bool, error) {
-	// TODO: Placeholder for future PR implementation
-	// https://github.com/kubermatic/bulward/pull/27
-	return true, nil
+func (o *OrganizationREST) isVisible(ctx context.Context, organization *Organization) (bool, error) {
+	attrs, err := filters.GetAuthorizerAttributes(ctx)
+	if err != nil {
+		return false, err
+	}
+	user := attrs.GetUser()
+	if user == nil {
+		klog.Warning("unknown user, you may running API extension server with --delegated-auth=false")
+		return true, nil
+	}
+
+	for _, sub := range append(
+		organization.Status.Members,
+		// This is important for seeing organizations you own before controller synces status
+		// otherwise a watch misses create event
+		organization.Spec.Owners...,
+	) {
+		switch sub.Kind {
+		case rbacv1.UserKind:
+			if sub.Name == user.GetName() {
+				return true, nil
+			}
+		case rbacv1.GroupKind:
+			for _, grp := range user.GetGroups() {
+				if sub.Name == grp {
+					return true, nil
+				}
+			}
+		case rbacv1.ServiceAccountKind:
+			if fmt.Sprintf("system:serviceaccount:%s:%s", sub.Namespace, sub.Name) == user.GetName() {
+				return true, nil
+			}
+		default:
+			klog.Warning("unknown subject kind:" + sub.Kind + ", in organization " + organization.Name + " members, skipping")
```

Comment:
the logger isn't injected because the API extension server is using klog, thus for this concrete process, I'd stick with klog over `logr.Logger` 

mixnig two loggers with different logging styles (legacy line based & jsonlines) seems a bit bad omen.

Reply:
got it, makes sence